### ISSUE TITLE
Metaviral spades and Iontorrent?

### ISSUE THREAD

**Issue by CEJ-R** [ablab/spades#1081]
### Description of bug

When trying to assemble DNA viral samples using the flags `--metaviral` and `--iontorrent`, spades won't generate any contigs, even when running with positive controls which should generate decent output. Spades generates warnings saying that it cannot use single reads:
```
======= SPAdes pipeline finished WITH WARNINGS!

=== Error correction and assembling warnings:
 * 0:23:23.144  3198M / 13G   WARN    General                 (pair_info_count.cpp       : 187)   Single reads are not used in metagenomic mode
 * 0:23:23.145  3198M / 13G   WARN    General                 (repeat_resolving.cpp      :  81)   Insert size was not estimated for any of the paired libraries, repeat resolution module will not run.
```
Since `--rnaviral` works fine with `--iontorrent` for RNA samples this appears odd, also since it seems that the assembly generates decent output up to the stage where the warning occurs. 

So, is it true that only the one mode supports Iontorrent single reads, or am I missing something else? 

### spades.log

[spades.log](https://github.com/ablab/spades/files/10730153/spades.log)

### params.txt

[params.txt](https://github.com/ablab/spades/files/10730169/params.txt)

### SPAdes version

3.15.5

### Operating System

Linux-4.18.0-425.3.1.el8.x86_64-x86_64-with-redhat-8.7-Ootpa

### Python Version

3.6.8

### Method of SPAdes installation

Binaries

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment]
Assigning to metaviralSPAdes author

**Comment by Dmitry-Antipov** [comment]
All SPAdes -meta based pipelines do not use single reads for repeat resolution. Possibly this limitation should be removed, especially for ionTorrent - I'm not sure (actually never seen meta iontorrent data). SInce metaviral pipeline outputs contigs of some specific type after repeat resolution, it will fail with ionTorrent. 


If you are assembling viral samples (not metagenomes), likely the better option is not --metaviral but just regular SPAdes with minor parameter tuning - see https://github.com/ablab/spades/issues/1079 for details

**Comment by CEJ-R** [comment]
Hi @Dmitry-Antipov, and thank you for your reply!

Yes, I consider them to be metagenomic samples as the purpose is to determine whether we have indications of particular viruses in samples that otherwise contain bacteria (and possibly other genomes). This has worked excellent using the `--rnaviral` mode for RNA samples on our Iontorrent.

Do I understand you correctly then then that `--metaviral` won't/can't work for Iontorrent?
Could regular SPAdes, tuned, give a similar result to what we see with `--rnaviral`, or what would you suggest? (it seems you say this is mostly useful for non-metagenomic samples though...)

Thank you again!

**Comment by Dmitry-Antipov** [comment]
>Do I understand you correctly then then that --metaviral won't/can't work for Iontorrent?
Yes, currently with iontorrent you'll have really suboptimal results for metaSPAdes and none for metaviral.

This is not a technology limitation but SPAdes design and fix seems to be quite straightforward. 
Will you be able to compile SPAdes from source if we'll provide you a branch with fix?

**Comment by CEJ-R** [comment]
I am sure that we can find a way to install it if you supply a branch to try this. We also have a control-sample that we could share if you'd need something to test the fix with. 
Since the fastq is quite large, even trimmed, is there a way you would prefer us to share it with you?

**Comment by Dmitry-Antipov** [comment]
Likely globus option is not available (unless you have your own server in your institution). Possibly just google drive?

**Comment by CEJ-R** [comment]
We do have a server, if that would help. However, a drive would work too, and would probably be simpler for all. Do you have your own drive I could upload the file to, or an email to which I could share a link of my own?

**Comment by Dmitry-Antipov** [comment]
Better to attach to email, [spades.support@cab.spbu.ru](mailto:spades.support@cab.spbu.ru)

Thank you!

**Comment by CEJ-R** [comment]
I apologize for the delay, but I just sent a link to the address you gave @Dmitry-Antipov, containing three positive controls which should be a good basis for testing.

Please let me know if there's anything more I can do to help (which I'll do in a more timely maner)!
